tests/integration/local/start_lambda/test_start_lambda.py

```diff
@@ -1737,3 +1741,148 @@ def test_invoke_with_data_custom_invoke_images(self):
         self.assertEqual(response.get("Payload").read().decode("utf-8"), '"This is json data"')
         self.assertIsNone(response.get("FunctionError"))
         self.assertEqual(response.get("StatusCode"), 200)
+
+
+class TestFunctionNameFiltering(StartLambdaIntegBaseClass):
+    """Test function name filtering feature for start-lambda command"""
+
+    template_path = "/testdata/invoke/template.yml"
+
+    def setUp(self):
+        self.url = f"http://127.0.0.1:{self.port}"
+        self.lambda_client = boto3.client(
+            "lambda",
+            endpoint_url=self.url,
+            region_name="us-east-1",
+            use_ssl=False,
+            verify=False,
+            config=Config(signature_version=UNSIGNED, read_timeout=120, retries={"max_attempts": 0}),
+        )
+
+    @pytest.mark.flaky(reruns=3)
+    @pytest.mark.timeout(timeout=300, method="thread")
+    def test_invoke_function_success(self):
+        """Test invoking a function succeeds"""
+        response = self.lambda_client.invoke(FunctionName="EchoEventFunction", Payload='"test data"')
+        self.assertEqual(response.get("StatusCode"), 200)
+        self.assertEqual(response.get("Payload").read().decode("utf-8"), '"test data"')
+        self.assertIsNone(response.get("FunctionError"))
+
+
+class TestFunctionNameFilteringWithFilter(StartLambdaIntegBaseClass):
+    """Test function name filtering with specific functions"""
+
+    template_path = "/testdata/invoke/template.yml"
+    function_logical_ids = ["EchoEventFunction", "HelloWorldServerlessFunction"]
+
+    def setUp(self):
+        self.url = f"http://127.0.0.1:{self.port}"
+        self.lambda_client = boto3.client(
+            "lambda",
+            endpoint_url=self.url,
+            region_name="us-east-1",
+            use_ssl=False,
+            verify=False,
+            config=Config(signature_version=UNSIGNED, read_timeout=120, retries={"max_attempts": 0}),
+        )
+
+    @pytest.mark.flaky(reruns=3)
+    @pytest.mark.timeout(timeout=300, method="thread")
+    def test_invoke_filtered_function(self):
+        """Test invoking a function that is in the filter"""
+        response = self.lambda_client.invoke(FunctionName="EchoEventFunction", Payload='"filtered test"')
+        self.assertEqual(response.get("StatusCode"), 200)
+        self.assertEqual(response.get("Payload").read().decode("utf-8"), '"filtered test"')
+        self.assertIsNone(response.get("FunctionError"))
+
+    @pytest.mark.flaky(reruns=3)
+    @pytest.mark.timeout(timeout=300, method="thread")
+    def test_invoke_non_filtered_function(self):
+        """Test invoking a function that is NOT in the filter returns ResourceNotFoundException"""
+        with self.assertRaises(ClientError) as context:
+            self.lambda_client.invoke(FunctionName="RaiseExceptionFunction")
```

Comment:
It's very not-important, but it feels weird that you're calling the function that raises an exception. It looks like this is the exception that you're waiting for. I think it's clearer if you use a different function name.

Reply:
Ah, but in this case it was better before. It was good that it was a function that existed in the template. I'm saying that you should use a function that would normally work, to test that you didn't , like `HelloWorldLambdaFunction` or `FunctionWithMetadata` or something like that.